src/NuGet.Core/NuGet.ProjectManagement/Utility/BuildIntegratedProjectUtility.cs

```diff
@@ -41,7 +41,16 @@ public static class BuildIntegratedProjectUtility
         /// </summary>
         public static async Task<LockFile> GetLockFileOrNull(BuildIntegratedNuGetProject buildIntegratedProject)
         {
-            var lockFilePath = await buildIntegratedProject.GetAssetsFilePathAsync();
+            string lockFilePath;
+            try
+            {
+                lockFilePath = await buildIntegratedProject.GetAssetsFilePathAsync();
+            }
+            catch (ProjectNotLoadedException)
+            {
+                return null;
```

Comment:
If the lock file is corrupted or the lock file does not exist, we will have the same behavior. When the nomination and restore happens, the solution spec hash will change meaning the script execution will occur again.